android/services/mongodb-remote-perftests/src/androidTest/java/com/mongodb/stitch/android/services/mongodb/performance/SyncR2LOnlyPerformanceTestDefinitions.kt

```diff
@@ -0,0 +1,269 @@
+                    val numSyncedIds = Tasks.await(sync.syncedIds).size
+                    val numLocalDocs = Tasks.await(sync.count())
+
+                    SyncPerformanceTestUtils.assertIntsAreEqualOrThrow(
+                        numSyncedIds, numDocs, "Number of Synced Ids")
+                    SyncPerformanceTestUtils.assertIntsAreEqualOrThrow(
+                        numLocalDocs.toInt(), numDocs, "Number of Local Documents")
+
+                    // Disconnect the DataSynchronizer and wait
+                    // for the underlying streams to close
+                    ctx.testNetworkMonitor.connectedState = false
+                    while (ctx.testDataSynchronizer.areAllStreamsOpen()) {
+                        testHarness.logMessage("waiting for streams to close")
+                        Thread.sleep(1000)
+                    }
+
+                    // Randomly sample a percentage of the documents
+                    // that will be locally updated
+                    val shuffledDocIds = ids.shuffled()
+                    val numChangedDocs = Math.round(pctOfDocsWithChangeEvents*numDocs).toInt()
+                    val docsToUpdate =
+                        if (pctOfDocsWithChangeEvents > 0.0)
+                            shuffledDocIds.subList(0, numChangedDocs)
+                        else
+                            emptyList()
+
+                    val updateResult = Tasks.await(ctx.testColl.updateMany(
+                        Document("_id", Document("\$in", docsToUpdate)),
+                        Document("\$set", Document("newField", "blah"))
+                    ))
+
+                    // Assert that the update worked
+                    assertIntsAreEqualOrThrow(updateResult.matchedCount.toInt(), numChangedDocs,
+                        "RemoteUpdateResult.matchedCount")
+                    assertIntsAreEqualOrThrow(updateResult.modifiedCount.toInt(), numChangedDocs,
+                        "RemoteUpdateResult.modifiedCount")
+
+                    val numDocsChangedRemotely = Tasks.await(ctx.testColl.count(
+                        Document("newField", Document("\$exists", true))
+                    ))
+                    SyncPerformanceTestUtils.assertIntsAreEqualOrThrow(
+                        numChangedDocs, numDocsChangedRemotely.toInt(), "Remote document updates"
+                    )
+
+                    numberOfChangedDocs = numChangedDocs
+                },
+                testDefinition = { ctx, _, _ ->
+                    // Do the sync pass that will sync the remote changes to the local collection
+                    // Perform syncPass() and halt the test if the pass fails
+                    if (!SyncPerformanceTestUtils.doSyncPass(ctx)) {
+                        error("sync pass failed")
+                    }
+
+                    if (!SyncPerformanceTestUtils.doSyncPass(ctx)) {
```

Comment:
[minor] you should not need to run the sync pass twice here. What happens if you don't disconnect the network in the test setup?